framework/grid/GridView.php

```diff
@@ -272,6 +272,11 @@ public function init()
             $this->filterRowOptions['id'] = $this->options['id'] . '-filters';
         }
 
+        $id = $this->options['id'];
+        $options = Json::htmlEncode($this->getClientOptions());
+        $view = $this->getView();
+        $view->registerJs("jQuery('#$id').yiiGridView($options);");
+
```

Comment:
Could you elaborate, please, why did you move this code from `run()` to init?
I'm not sure that it's a good idea.